extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -776,6 +776,11 @@ public Supplier<Boolean> apply(ClassInfo classInfo) {
                         // to share method and annotations between Resource classes
                         continue;
                     }
+
+                    if (resourceClassImplCount == 0 && index.getKnownDirectImplementors(classInfo.name()).isEmpty()
```

Comment:
`getKnownDirectImplementors` returns 1 implementor, `getAllKnownImplementors` returns 0 implementors. Want to debug this before the pr is merged.

Reply:
I would add a comment to mention we skip them because they are REST Clients.

Reply:
> `getKnownDirectImplementors` returns 1 implementor, `getAllKnownImplementors` returns 0 implementors. Want to debug this before the pr is merged.

javadoc of `getKnownDirectImplementors` states that this is expected.
> The list of implementors also includes direct subinterfaces. This is inconsistent with getAllKnownImplementors(DotName), which doesn't return subinterfaces.
Note that this will only pick up classes that directly implement given interface. It will not pick up classes implementing subinterfaces.

Reply:
I don't understand why it's fine to use `getKnownDirectImplementors`, then?

Isn't it fine to only test `resourceClassImplCount`?